app/models/miq_server/log_management.rb

```diff
@@ -16,7 +16,7 @@ def format_log_time(time)
     time.respond_to?(:strftime) ? time.strftime("%Y%m%d_%H%M%S") : "unknown"
   end
 
-  def post_historical_logs(taskid)
+  def post_historical_logs(taskid, context)
```

Comment:
Same as below, we can use dependency injection to pass any kind of depot instead of the context here... it will make writing tests here easier.